From fb7fcaad1f53ae6e81915aad7090fcfa08f17e2d Mon Sep 17 00:00:00 2001
From: Robert Marko <robimarko@gmail.com>
Date: Sat, 10 May 2025 20:44:51 +0200
Subject: [PATCH] main: use remove_new

Kernels 6.1 and newer support using void based .remove_new op
instead, so use it as its compatible with 6.1 onwards.

Signed-off-by: Robert Marko <robimarko@gmail.com>
---
 nss_dp_main.c | 11 ++++++++++-
 1 file changed, 10 insertions(+), 1 deletion(-)

--- a/nss_dp_main.c
+++ b/nss_dp_main.c
@@ -968,7 +968,11 @@ fail:
  * Note: We only remove the physical ports here. Virtual
  * port devices are removed explicitly by the VP module.
  */
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+static void nss_dp_remove(struct platform_device *pdev)
+#else
 static int nss_dp_remove(struct platform_device *pdev)
+#endif
 {
 	uint32_t i;
 	struct nss_dp_dev *dp_priv;
@@ -1004,8 +1008,9 @@ static int nss_dp_remove(struct platform
 		free_netdev(dp_priv->netdev);
 		dp_global_ctx.nss_dp[i] = NULL;
 	}
-
+#if (LINUX_VERSION_CODE < KERNEL_VERSION(6, 1, 0))
 	return 0;
+#endif
 }
 
 static struct of_device_id nss_dp_dt_ids[] = {
@@ -1016,7 +1021,11 @@ MODULE_DEVICE_TABLE(of, nss_dp_dt_ids);
 
 static struct platform_driver nss_dp_drv = {
 	.probe = nss_dp_probe,
+#if (LINUX_VERSION_CODE >= KERNEL_VERSION(6, 1, 0))
+	.remove_new = nss_dp_remove,
+#else
 	.remove = nss_dp_remove,
+#endif
 	.driver = {
 		   .name = "nss-dp",
 		   .owner = THIS_MODULE,
